TombEngine/Objects/TR3/Vehicles/minecart.cpp

```diff
@@ -923,6 +951,22 @@ namespace TEN::Entities::Vehicles
 
 			MinecartToEntityCollision(minecartItem, laraItem);
 		}
+		else if (laraItem->HitPoints <= 0 && 
+			laraItem->Animation.ActiveState != MINECART_STATE_DEATH && 
+			laraItem->Animation.ActiveState != MINECART_STATE_TURN_DEATH &&
+			laraItem->Animation.ActiveState != MINECART_STATE_WALL_DEATH)
+		{
+			laraItem->Animation.AnimNumber = Objects[ID_MINECART_LARA_ANIMS].animIndex + MINECART_ANIM_DEATH;
```

Comment:
You can try to use `SetAnimation()` function instead of doing manual assignments of all `Animation` fields here.

Reply:
I thought `SetAnimation()` didn't work for assigning animations from other slots?

Reply:
Oh, that's right, possibly we should provide an overload for that.